### PR TITLE
Gradient boost survival function for GradientBoostingSurvivalAnalysis

### LINKED ISSUE #121: Calculating the survival function for the predicted risks from GradientBoostingSurvivalAnalysis(loss = 'coxph')
Hello, 

I am trying to predict the survival function for the datapoints in my test set. I understand that the survival function is implemented for the CoxPH model. Below, I am comparing how I estimate the survival function using the baseline estimator. I am able to reproduce the survival curve below. 

```
import numpy as np

from sksurv.ensemble import GradientBoostingSurvivalAnalysis
from sksurv.linear_model import CoxPHSurvivalAnalysis
from sksurv.linear_model.coxph import BreslowEstimator
import sksurv

coxph = CoxPHSurvivalAnalysis()
coxph_fit = coxph.fit(X_full_train, y_train)

risk_score_coxph = coxph_fit.predict(X_full_test) #raw scores

baseline_estimator = BreslowEstimator()

coxph_train_risks = coxph_fit.predict(X_full_train)
_, event, time = sksurv.util.check_arrays_survival(X_full_train, y_train)
surv_func_coxph = baseline_estimator.fit(coxph_train_risks, event, time).get_survival_function(coxph.predict(X_full_test))

surv_func_coxph_true = coxph.predict_survival_function(X_full_test) #what is implemented in the library

for true, my_implementation in zip(surv_func_coxph_true, surv_func_coxph):
    y_true = true.y
    y_implement = my_implementation.y
    
    assert (np.allclose(y_true, y_implement))

```

Now, I want to predict the survival functions from the predicted risks of the GradientBoostingSurvivalAnalysis. Do I follow the same path as above and use the baseline estimator for the outputs of the boosted cox model? 

```
estimator = GradientBoostingSurvivalAnalysis(loss = 'coxph')
estimator_fitted = estimator.fit(X_full_train, y_train)

risk_score_coxboost = estimator_fitted.predict(X_full_test) #raw scores

baseline_estimator = BreslowEstimator()

coxboost_train_risks = estimator_fitted.predict(X_full_train)
_, event, time = sksurv.util.check_arrays_survival(X_full_train, y_train)
surv_func_coxboost = baseline_estimator.fit(coxboost_train_risks, event, time).get_survival_function(estimator_fitted.predict(X_full_test))

```

### PR DESCRIPTION
**Pull request by jlim13**
**Checklist**

- [x] closes #121 
- [x] py.test passes
- [x] tests are included
- [x] code is well formatted
- [x] documentation renders correctly

**What does this implement/fix? Explain your changes**

### CODE REVIEW

**Comment by jlim13** [comment]
Ah wait I see that this function was already added. woops, my bad. the version i pulled a few days ago didn't have these functions

**Comment by sebp** [comment]
Yes, I just added it this weekend. Sorry that that you're work is 
redundant now. In case you want to work on something else in the future, 
please open an issue mentioning what you are going to work on to avoid 
duplicate work.
